pkg/kube/cluster-init.sh

```diff
@@ -183,72 +338,210 @@ trigger_k3s_selfextraction() {
         /usr/bin/k3s check-config >> $INSTALL_LOG 2>&1
 }
 
+# wait for debugging flag in /persist/k3s/wait_{flagname} if exist
+wait_for_item() {
+        filename="/persist/k3s/wait_$1"
+        processname="k3s server"
+        while true; do
+                if [ -e "$filename" ]; then
```

Comment:
isn't it possible to check directly in the condition of the while loop that `$filename` exists instead of creating an endless loop and then inside check if the file exists and break if it does not?

Reply:
certainly yes

Reply:
```sh
until [ -f "$filename" ]; do
...
done
```
is what are you looking for...